Apps/Admin/Tests/Functional/cypress/integration/e2e/patient-details.cy.js

```diff
@@ -188,3 +192,35 @@ describe("Patient details message verification", () => {
         });
     });
 });
+
+describe("Patient details page as reviewer", () => {
+    beforeEach(() => {
+        cy.login(
+            Cypress.env("keycloak_reviewer_username"),
+            Cypress.env("keycloak_reviewer_password"),
+            "/support"
+        );
+    });
+
+    // verify that the reviewer cannot use the block access controls
+    it("Verify block access readonly", () => {
+        performSearch("HDID", hdid);
+        cy.get("[data-testid=user-table]")
+            .find("tbody tr.mud-table-row")
+            .first()
+            .click();
+
+        cy.get(`[data-testid*="block-access-switch-"]`).each(($el) => {
+            // follow the mud tag structure to find the mud-readonly class
+            cy.wrap($el).parent().parent().should("have.class", "mud-readonly");
+        });
+
+        // Clicke any switch and check if the dirty state has exposed the save and cancel buttons
+        cy.get(`[data-testid=block-access-switch-${switchName}]`)
+            .should("exist")
+            .click();
+
+        cy.get("[data-testid=block-access-cancel]").should("not.exist");
```

Comment:
The cancel and save checks will always be 'not.exist' unless a dataset item has been selected i.e., they do not show up until a dataset is selected.  You need to select a dataset before checking if buttons exist or not.